pathplannerlib/src/main/native/include/pathplanner/lib/util/swerve/SwerveSetpointGenerator.h

```diff
@@ -0,0 +1,128 @@
+#include "pathplanner/lib/config/RobotConfig.h"
+#include "pathplanner/lib/util/DriveFeedforwards.h"
+#include "pathplanner/lib/util/swerve/SwerveSetpoint.h"
+
+#include <frc/kinematics/SwerveModuleState.h>
+#include <frc/kinematics/SwerveDriveKinematics.h>
+#include <frc/RobotController.h>
+
+using namespace pathplanner;
+
+/**
+ * Swerve setpoint generator based on a version created by FRC team 254.
+ *
+ * <p>Takes a prior setpoint, a desired setpoint, and outputs a new setpoint that respects all the
+ * kinematic constraints on module rotation and wheel velocity/torque, as well as preventing any
+ * forces acting on a module's wheel from exceeding the force of friction.
+ */
+class SwerveSetpointGenerator {
+public:
+
+	/**
+	 * Create a new swerve setpoint generator
+	 */
+	SwerveSetpointGenerator();
+
+	/**
+	 * Create a new swerve setpoint generator
+	 *
+	 * @param config The robot configuration
+	 * @param maxSteerVelocity The maximum rotation velocity of a swerve module, in radians
+	 *     per second
+	 */
+	SwerveSetpointGenerator(RobotConfig *config,
```

Comment:
Why take in RobotConfig as a pointer? I believe the rest of the library uses `const &` when dealing with config.

Reply:
Right, that's an unnecessary pointer right there. Just a question due to the RobotConfig not having a default constructor, should I pass it to the generateSetpoint function rather than having it as an attribute and pass it through the constructor of the class.

Reply:
You could probably just add a default constructor to RobotConfig